src/cassandra/value.rs

```diff
@@ -496,4 +496,22 @@ impl Value {
         };
         unsafe { cass_value_get_uuid(self.0, &mut output).to_result(Uuid::build(output)) }
     }
+
+    /// Get this value as a BigDecimal
+    pub fn get_decimal(&self) -> Result<BigDecimal> {
+        let mut varint = std::ptr::null();
+        let mut varint_size = 0;
+        let mut scale = 0;
+        let slice = unsafe {
+            if cass_value_get_decimal(self.0, &mut varint, &mut varint_size, &mut scale)
+                != cassandra_cpp_sys::CassError::CASS_OK
+            {
+                return Err(CASS_ERROR_LIB_INVALID_VALUE_TYPE.to_error());
+            }
+            slice::from_raw_parts(varint, varint_size)
+        };
+        let bigint = BigInt::from_signed_bytes_be(slice);
```

Comment:
Sigh, I can't find any actual docs that confirm the byte ordering and format. Presumably you've tested some values to confirm it's OK?

Reply:
Decimal is derived from [java.math.BigDecimal](http://docs.oracle.com/javase/7/docs/api/java/math/BigDecimal.html) from here: https://docs.datastax.com/en/cql-oss/3.x/cql/cql_reference/cql_data_types_c.html
From that text, I assume integer part is https://docs.oracle.com/javase/7/docs/api/java/math/BigInteger.html which use bigendian order.

Reply:
Brilliant, thanks - yes that looks good. I'm happy with this.